### ISSUE TITLE
Local Refs not resolving properly

### ISSUE THREAD

**Issue by seanarnold** [davishmcclurg/json_schemer#146]
I think this was a breaking change between 1.x and 2.x, and the Changlog indicates I probably have to do something with my `ref_resolver` but I'm not quite sure what.

Take this example schema:

`lib/schemas/user.json`:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "https://example.com/project/-/raw/main/json-schemas/user-model.json",
  "$ref": "#/definitions/UserModel",
  "definitions": {
    "UserModel": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string"
        }
      }
    }
  }
}
```

In V1 we're calling and validating against it like so:

```ruby
schemer = JSONSchemer.schema('lib/schemas/user.json')
hash = {"name" => 1234}
schemer.valid?(hash)
=> false
```

However in V2 we're getting this error
```ruby
schemer = JSONSchemer.schema('lib/schemas/user.json')
hash = {"name" => 1234}
schemer.valid?(hash)

[2] pry(#<RSpec::Matchers::DSL::Matcher>)> schemer.valid?(hash)
KeyError: key not found: #<URI::File file:///Users/seanarnold/projects/proj/path...
from /Users/seanarnold/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/json_schemer-2.0.0/lib/json_schemer/schema.rb:195:in `fetch'
```

I'd appreciate some help with adjusting my config to get these relative refs working <3

**Comment by davishmcclurg** [comment]
Hi @seanarnold—thanks for opening this. I think I've got a fix that you can try out: https://github.com/davishmcclurg/json_schemer/pull/147

You should note that draft 7 ignores sibling keywords (except `definitions`) when `$ref` if present (`$id` is ignored in your example). More info:
- https://github.com/davishmcclurg/json_schemer/issues/44#issuecomment-1684675824
- https://github.com/json-schema-org/JSON-Schema-Test-Suite/pull/493
- https://github.com/json-schema-org/JSON-Schema-Test-Suite/pull/458

Let me know if you can try out the fix.

**Comment by seanarnold** [comment]
Thanks, that seems to fix it for us :) Any chance 2.1.0 could be released shortly? Thank you!

**Comment by davishmcclurg** [comment]
> Thanks, that seems to fix it for us :) Any chance 2.1.0 could be released shortly? Thank you!

Fix just went out in 2.1.0
